pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -625,6 +628,7 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusterForPortOrUDS(pluginPara
 			string(instance.Service.Hostname), "", pluginParams.Node.DNSDomain, instance.Endpoint.ServicePort)
 	}
 	setUpstreamProtocol(localCluster, instance.Endpoint.ServicePort)
+	setProtocolSelection(pluginParams.Node, localCluster, instance.Endpoint.ServicePort, model.TrafficDirectionInbound)
```

Comment:
what is the difference between this and the previous function?

Reply:
`setUpstreamProtocol` vs `setProtocolSelection`? I added a direction in `setProtocolSelection`. We can definitely change the previously one. Either one is okay with me.

Reply:
More like, can you merge this into the setUpstreamProtocol function ?